File: 01.Python/1800.py
from typing import List

class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        currentSum = nums[0]
        maxSum = nums[0]

        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                currentSum += nums[i]
            else:
                currentSum = nums[i]
            
            maxSum = max(maxSum, currentSum)
        
        return maxSum

sol = Solution()

nums = [10, 20, 30, 5, 10, 50]
result = sol.maxAscendingSum(nums)

print(result)

File: 03.CPP/1800.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int maxAscendingSum(std::vector<int>& nums) {
        int currentSum = nums[0];
        int maxSum = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i - 1]) {
                currentSum += nums[i];
            } else {
                currentSum = nums[i];
            }

            maxSum = std::max(maxSum, currentSum);
        }

        return maxSum;
    }
};

int main() {
    Solution sol;

    std::vector<int> nums = {10, 20, 30, 5, 10, 50};
    int result = sol.maxAscendingSum(nums);

    std::cout << result << "\n";
    return 0;
}